pkg/client/elasticsearch/client.go

```diff
@@ -170,30 +172,30 @@ func (mc *MetricsClient) GetMetricByName(name types.NamespacedName, info provide
 func (mc *MetricsClient) GetMetricBySelector(namespace string, selector labels.Selector, info provider.CustomMetricInfo, metricSelector labels.Selector) (*custom_metrics.MetricValueList, error) {
 	t, ctx := tracing.NewTransaction(context.TODO(), mc.tracer, "elasticsearch-provider", "GetMetricBySelector")
 	defer tracing.EndTransaction(t)
-	klog.Infof("-> elasticsearchProvider.GetMetricBySelector(namespace=%v,selector=%v,info=%v,metricSelector=%v)", namespace, selector, info, metricSelector)
+	logger.V(1).Info("GetMetricBySelector", "namespace", namespace, "selector", selector, "info", info.String(), "metricSelector", metricSelector)
 	return mc.metricsFor(&ctx, namespace, selector, info, metricSelector)
 }
 
 func (mc *MetricsClient) GetExternalMetric(
 	_, _ string,
 	_ labels.Selector,
 ) (*external_metrics.ExternalMetricValueList, error) {
-	klog.Error("GetExternalMetric: external are not supported by Elasticsearch metrics client")
+	logger.V(2).Info("GetExternalMetric: not supported by Elasticsearch metrics client")
```

Comment:
Should not stay an error log?

Reply:
I changed it to align with the other same "unsupported" log present in ListExternalMetrics. But yes, I now think I understand why it was set on error. Because ListExternalMetrics return a nil map of metrics, GetExternalMetric should not be called here. I will revert.